utility.py

```diff
@@ -0,0 +1,135 @@
+    def distribution_computation(self, name):
+        # create X-axis and Y-axis in order to plot histograms
+        """
+        X - Counts (or other measurements)
+        Y - # of cells that has that counts/value (Frequency)
+        """
+        statistics_result = dict((i, list(self.hist).count(i)) for i in list(self.hist))
+        statistics_result = dict(sorted(statistics_result.items()))
+        count_0_pair = []
+        if 0 in statistics_result:
+            count_0_pair = [0, statistics_result[0]]
+        
+        statistics_result.pop(0, None)
+
+        x_axis = list(statistics_result.keys())
+
+        fig, ax = plt.subplots()
+        ax.bar(range(len(statistics_result)), list(statistics_result.values()), align='center')
+        plt.xticks((0, len(statistics_result) - 1), (x_axis[0], x_axis[len(statistics_result) - 1]))
+        plt.xlabel('Counts (or other measurements)')
+        plt.ylabel('Frequency (# of grids has that counts)')
+        fig.savefig(name)
+
+        return statistics_result, x_axis, count_0_pair, len(statistics_result)
+    # =====================================
+    # Write out geojson file
+    # Polygon Format:
+    # [ [
+    #      [-180.0, 79.1713346],
+    #      [-180.0, 85.0511288],
+    #      [-135.0, 85.0511288],
+    #      [-135.0, 79.1713346]
+    #   ]
+    # ]
+    def geojson_write(self, level_val, bounding_box_collec, directory_path, cell_num, initial_area, in_grid_ids, kd_tree_mode, flag_val):
+        # declare variables
+        json_dic = {}
+        feature_list = []
+        json_dic['type'] = 'FeatureCollection'
+
+        # add all cells and counts into the feature list
+        for index in range(len(self.hist)):
+            tmp_dic = {}
+            geometry_dic = {}
+            properties_dic = {}
+            geometry_dic['type'] = 'Polygon'
+            geometry_dic['coordinates'] = [[ [bounding_box_collec[index][0],bounding_box_collec[index][1]],
+                                         [bounding_box_collec[index][0],bounding_box_collec[index][3]],
+                                         [bounding_box_collec[index][2],bounding_box_collec[index][3]],
+                                         [bounding_box_collec[index][2],bounding_box_collec[index][1]] ]]
+        
+            properties_dic['counts'] = self.hist[index]
+            if flag_val:
+                if in_grid_ids is None:
+                    properties_dic['gridId'] = index + 1
+                else:
+                    properties_dic['gridId'] = in_grid_ids[index]
+
+            tmp_dic['type'] = 'Feature'
+            tmp_dic['geometry'] = geometry_dic
+            tmp_dic['properties'] = properties_dic
+            feature_list.append(tmp_dic)
+            del tmp_dic
+            del geometry_dic
+            del properties_dic
+        json_dic['features'] = feature_list
+
+        # save the dictionary structure as a Geojson file
+        if kd_tree_mode == 'tree_v1':
```

Comment:
Again, it'd be nice to completely separate the file logic from the data parsing logic.

Reply:
separated the file logic and the data-processing logic.